Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -181,11 +199,17 @@ BTDiagnostics::DoDump (int step, int i_buffer, bool force_flush)
 
 
 bool
-BTDiagnostics::DoComputeAndPack (int step, bool /*force_flush*/)
+BTDiagnostics::DoComputeAndPack (int step, bool force_flush)
 {
     // always set to true for BTDiagnostics since back-transform buffers are potentially
-    // computed and packed every timstep, except at initialization when step == -1.
-    return (step>=0);
+    // computed and packed every timstep, except at initialization when step == -1, or when
+    // force_flush is set to true.
+    if (step < 0 ) {
```

Comment:
Do you maybe want to change the logic to
```suggestion
    if (step < 0 || force_flush ) {
```
?

The comment reads a bit like they are of equal control, but at the moment `force_flush` is only queried if we are not in the init step. This might lead to surprises down the road?

Otherwise, if the current logic is intentional, we can simplify to:
```C++
if (step < 0 ) {
    return false;
} else if (force_flush) {
    return false;
} else {
    return true;
}
````

Reply:
the current logic is intentional. I will add in this simplification